signingscript/src/signingscript/script.py

```diff
@@ -75,6 +84,35 @@ def get_default_config(base_dir=None):
     return default_config
 
 
+def write_apple_notarization_credentials(context):
+    """Writes the notarization credential to a file
+
+    Adds property to context: apple_credentials_path
+
+    Args:
+        context: Running task Context
+    """
+    cert_type = task_cert_type(context)
+    apple_notarization_configs = load_apple_notarization_configs(context.config["apple_notarization_configs"])
+    if cert_type not in apple_notarization_configs:
+        raise SigningScriptError("Credentials not found for scope: %s" % cert_type)
+    scope_credentials = apple_notarization_configs.get(cert_type)
+    if len(scope_credentials) != 1:
+        raise SigningScriptError("There should only be 1 scope credential, %s found." % len(scope_credentials))
+
+    context.apple_credentials_path = os.path.join(
+        os.path.dirname(context.config["apple_notarization_configs"]),
+        'apple_api_key.json',
+    )
+    if os.path.exists(context.apple_credentials_path):
```

Comment:
This code path means that this function doesn't do what it claims ("write credentials"), and also doesn't throw an error. Why does it exist, and what can we do to improve this? Is it a remnant from when this was being called per task?

Reply:
Basically it will write the first time, and skip for any subsequent tasks.
The function still checks for scope issues, so there's a reason for it to exist.
Any specific suggestions?

Reply:
In that case, it ought to have a different name, or move the scope checks out :).